### PR TITLE
API-165_API_182: Get a single currency and get list of currencies

### PR DESCRIPTION
**Pull request by momoss**
**Description (for Contributor and Core Developer)**

Get a single currency and get list of currencies

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | Ok
| Added Behats                      | -
| Added integration tests           | Ok
| Changelog updated                 | -
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | Todo
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: src/Pim/Bundle/ApiBundle/Controller/CurrencyController.php [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+    /** @var ParameterValidatorInterface */
+    protected $parameterValidator;
+
+    /** @var PaginatorInterface */
+    protected $paginator;
+
+    /** @var array */
+    protected $apiConfiguration;
+
+    public function __construct(
```
Comment (skeleton) [inline src/Pim/Bundle/ApiBundle/Controller/CurrencyController.php]: phpdoc

File: src/Pim/Bundle/ApiBundle/Controller/CurrencyController.php [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+     * @AclAncestor("pim_api_currency_list")
+     */
+    public function getAction(Request $request, $code)
+    {
+        $currency = $this->repository->findOneByIdentifier($code);
+        if (null === $currency) {
+            throw new NotFoundHttpException(sprintf('Currency "%s" does not exist.', $code));
+        }
+
+        $currencyApi = $this->normalizer->normalize($currency, 'external_api');
```
Comment (a2xchip) [inline src/Pim/Bundle/ApiBundle/Controller/CurrencyController.php]: ? `normalizedCurrency`

Comment (momoss) [inline src/Pim/Bundle/ApiBundle/Controller/CurrencyController.php]: The naming is coherent with others API controller.

**Comment by momoss** [comment]
https://core-ci.akeneo.com/blue/organizations/jenkins/akeneo%2Fpim-community-dev/detail/PR-6203/3/pipeline

**Pull request merged**